editor/editor_file_system.cpp

```diff
@@ -2282,7 +2282,7 @@ void EditorFileSystem::reimport_files(const Vector<String> &p_files) {
 					pr.step(reimport_files[i].path.get_file(), i);
 					_reimport_file(reimport_files[i].path);
 				} else {
-					Ref<ResourceImporter> importer = ResourceFormatImporter::get_singleton()->get_importer_by_name(reimport_files[from].importer);
+					Ref<ResourceImporter> importer = ResourceFormatImporter::get_singleton()->get_importer_by_name(reimport_files[i].importer);					
```

Comment:
The WorkerThreadPool below still uses `from`, though it doesn't seem like a full fix?

Reply:
The `from` will be incremented at line 2310 but its only reached if we pass `ERR_CONTINUE(!importer.is_valid());`, but in the case of #84473, the first resource is corrupt, this line will fail, and `from` will not be incremented.
Maybe we need to refactor this function, because it has multiple indexes that are unncesseceraly.

Reply:
If the sorrounding code was correct, it wouldn't matter whether `i` or `from` is used as an index there. To me, the problem is that `from` is not tracked properly. On the one hand, there's the spot you mention, which would need to be changed to something like this:
```
if (!importer.is_valid()) {
    from = i + 1;
    ERR_CONTINUE(!importer.is_valid());
}
```

On the other hand, the first early-fail check needs an upgrade as well, to this:
```
if (groups_to_reimport.has(reimport_files[i].path)) {
    from = i + 1;
    continue;
}
```

A slight refactor making the head-tail more robust and readable wouldn't do any harm, but two indices (or a reasonable replacement) is still needed so the function can remember at which element the current import group starts.

I haven't tested this statements, so TIWAGOS.

Reply:
I believe RandomShaper is correct about that part of this.

One of the other unintuitive things about this code that isn't obvious is that `use_multiple_threads` is always true if the file's type is invalid, which can happen if you try to reimport a file with a bad file extension.

Mixing and matching file formats which support multiple threads and those which do not seems to cause that `(!importer.is_valid())` case, so I think that's one way to test it but when I hit this a few months ago,

I only hit this when I was using GDScript to force all of some unrelated files into a single reimport_files call, and had included the bad file (I think it was a .txt file, since technically those don't get imported).